### ISSUE TITLE
Feature request: "Open in IINA" system-wide contextual menu service

### ISSUE THREAD

**Issue by JayBrown** [lhc70000/iina#1753]
**System and IINA version:**

- OS X 10.11.6 (soon macOS 10.13.5)
- IINA 0.0.15.1

**Feature request:**
It would be great to have a system-wide Service in the contextual menu, e.g. "Open in IINA". For example, I have lots of YouTube channels in my RSS reader feed, and when a video interests me, I'd rather CTRL-click on the feed message title, which hides a href to the Youtube video's URL, and select "Open in IINA" from the contextual menu. Obviously this is not a functionality to use on actual video files, because with those you can just choose "Open" (if IINA is the default player) or "Open With". So this would be a Service that (primarily) grabs text strings, i.e URLs, e.g. remote paths to known video platforms like YouTube, Vimeo etc., but maybe also remote paths with basenames with known video extensions like .mp4, .mkv, .mov, .m4v, .m4a, .mp3 etc. It should also work with a list of several URLs. I know that users could script something like this with Automator, but maybe it's better to have a system-wide solution bundled with IINA.

**Comment by lhc70000** [comment]
I'm not sure whether it's technically possible though.

**Comment by saagarjha** [comment]
Possible in what sense? It shouldn't be hard to vend a service that takes a URL and opens it in IINA–actually, such a service would basically obsolete our browser extensions. Or are you talking about detecting media from the URL?

**Comment by EnricoMonese** [comment]
This is implemented now if I'm not mistaken?

**Comment by EnricoMonese** [comment]
As said in https://github.com/iina/iina/issues/2290#issuecomment-465231749 by @anohren there should be a way to find out about this. I think adding a section in the settings where it explains how to enable it would help!

**Comment by anohren** [comment]
Yes, some discoverablility would be good. I just stumbled upon it by looking for it after thinking "this should exist". Apparently it did.

The only place I can think of in settings where it wouldn't feel out of place would be next to the links to the browser extensions.
